chainerx_cc/chainerx/cuda/cuda_device/fill.cu

```diff
@@ -111,7 +111,20 @@ __global__ void SetVecInMat(
         int64_t mat_row_start,
         int64_t mat_col_start) {
     auto mat_it = mat_indexer.It(0);
-    for (auto vec_it = vec_indexer.It(blockIdx.x * blockDim.x + threadIdx.x, blockDim.x * gridDim.x); vec_it; ++vec_it) {
+    // If the array has more than 2^31 elements, the id calculation overflows
+    int64_t id = static_cast<int64_t>(blockIdx.x);
+    int64_t size = static_cast<int64_t>(gridDim.x);
+    int64_t thread_id = static_cast<int64_t>(threadIdx.x);
+    if (blockDim.x & (blockDim.x - 1)) {
```

Comment:
Why is this branching needed?

Reply:
To use bit shifting instead of int64 multiplication which is slow on GPUs

Reply:
Is that significant? The multiplication is only done once per thread.

Reply:
I will properly measure it,
My intution tells me that in ElementWise kernels of simple arithmetic operations it can increase execution time a lot.

Reply:
My intuition is opposite. Single int64 operation should be by far more lightweight than thread / global memory overhead.
Let's measure it and see :)

Reply:
Ugh, you are completely right 😂